project/app/api/submission.py

```diff
@@ -1,91 +1,252 @@
+        Function that takes the value of checksum and assert that it's 128
+        characters long to ensure that it's the proper length for a SHA512
+        checksum.
+    </p>
     """
+    SubmissionID: int = Field(..., example=265458)
+    URL: str = Field(
+        ...,
+        example=
+        "https://test-image-bucket-14579.s3.amazonaws.com/bucketFolder/1600554345008-lg.png"
+    )
+    Checksum: str = Field(
+        ...,
+        example=
+        "edbd2c0cd247bda620f9a0a3fe5553fb19606929d686ed3440742b1a25df426a8e6d3188b7eec163488764cc72d8cee67faba47e29f7744871d94d2a19dc70de"
+    )
+
+    @validator("SubmissionID")
+    def check_valid_subid(cls, value):
+        # no neg numbers and no int overflows
+        assert value >= 0
+        assert value < sys.maxsize
+        return value
+
+    @validator("Checksum")
+    def check_sha_len(cls, value):
+        assert len(value) == 128
+        return value
+
+
+@router.post("/submission/text")
+async def submission_text(sub: Submission):
+    """Takes a Submission Object and calls the Google Vision API to text annotate
+    the passed s3 link, then passes thos concatenated transcriptions to the SquadScore
+    method, returns:
+
+    ```
+    {"SubmissionID": int, "IsFlagged": boolean, "Complexity": int}
+    ```
+
+    """
+    transcriptions = ""
+    confidence_flags = []
+    # unpack links for files in submission object
+    for page_num in sub.Pages:
+        # re-init the sha algorithm every file that is processed
+        hash = sha512()
+        # fetch file from s3 bucket
+        r = get(sub.Pages[page_num]["URL"])
+        # update the hash with the file's content
+        hash.update(r.content)
+        try:
+            # assert that the has is the same as the one passed with the file
+            # link
+            assert hash.hexdigest() == sub.Pages[page_num]['Checksum']
+        except AssertionError:
+            # return some usefull information about the error including what
+            # caused it and the file effected
+            return JSONResponse(status_code=422,
+                                content={
+                                    "ERROR": "BAD CHECKSUM",
+                                    "file": sub.Pages[page_num]
+                                })
+        # add the response from google_api to a string with an ending
+        # line break and the confidence flag from the method that determins if
+        # the student is reminded about their handwritting
+        conf_flag, trans = await vision.transcribe(r.content)
+        transcriptions += trans + "\n"
+        confidence_flags.append(conf_flag)
+    # score the transcription using SquadScore algorithm
+    score = await squad_score(transcriptions, scaler)
+
+    # return the complexity score to the web team with the SubmissionID
+    return JSONResponse(
+        status_code=200,
+        content={
+            "SubmissionID": sub.SubmissionID,
+            "IsFlagged": False,
```

Comment:
What is the "IsFlagged" boolean for? It's different than the LowConfidence flag correct?